From: Hao Li <hao.li@linux.dev>
Date: Thu, 27 Aug 2026 15:18:06 +0800
Subject: [PATCH] mm: remove unused anon_vma_trylock_write()

The last user of anon_vma_trylock_write() was removed by cc22b9978509,
leaving the helper unused.  Remove it.

Link: https://lore.kernel.org/20260827071845.17636-1-hao.li@linux.dev
Signed-off-by: Hao Li <hao.li@linux.dev>
Reviewed-by: Lorenzo Stoakes (ARM) <ljs@kernel.org>
Acked-by: David Hildenbrand (Arm) <david@kernel.org>
Cc: Liam R. Howlett <liam@infradead.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/internal.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/mm/internal.h b/mm/internal.h
index 38b1165212c94..da833cafcd599 100644
--- a/mm/internal.h
+++ b/mm/internal.h
@@ -295,11 +295,6 @@ static inline void anon_vma_lock_write(struct anon_vma *anon_vma)
 	down_write(&anon_vma->root->rwsem);
 }
 
-static inline int anon_vma_trylock_write(struct anon_vma *anon_vma)
-{
-	return down_write_trylock(&anon_vma->root->rwsem);
-}
-
 static inline void anon_vma_unlock_write(struct anon_vma *anon_vma)
 {
 	up_write(&anon_vma->root->rwsem);
-- 
2.47.3

